packages/@aws-cdk/aws-cognito/lib/user-pool-attr.ts

```diff
@@ -141,6 +141,84 @@ export interface CustomAttributeConfig {
    * @default - None.
    */
   readonly numberConstraints?: NumberAttributeConstraints;
+
+  /**
+   * Specifies whether the attribute type is developer only. This attribute can only be modified by an administrator.
+   * Users will not be able to modify this attribute using their access token.
+   *
+   * @default false
+   */
+  readonly developerOnly?: boolean
+
+  /**
+   * Specifies whether the value of the attribute can be changed.
+   * For any user pool attribute that's mapped to an identity provider attribute, you must set this parameter to true.
+   * Amazon Cognito updates mapped attributes when users sign in to your application through an identity provider.
+   * If an attribute is immutable, Amazon Cognito throws an error when it attempts to update the attribute.
+   *
+   * @default false
+   */
+  readonly mutable?: boolean
+
+}
+
+/**
+ * Constraints that can be applied to a custom attribute of string type.
+ */
+export interface BaseCustomAttributeProps {
+  /**
+   * Specifies whether the attribute type is developer only. This attribute can only be modified by an administrator.
+   * Users will not be able to modify this attribute using their access token.
+   *
+   * @default false
+   */
+  readonly developerOnly?: boolean
+
+  /**
+   * Specifies whether the value of the attribute can be changed.
+   * For any user pool attribute that's mapped to an identity provider attribute, you must set this parameter to true.
+   * Amazon Cognito updates mapped attributes when users sign in to your application through an identity provider.
+   * If an attribute is immutable, Amazon Cognito throws an error when it attempts to update the attribute.
+   *
+   * @default false
+   */
+  readonly mutable?: boolean
+}
+
+/**
+ * The base class for all custom attribute types.
+ * All common properties are set here and the method `baseAttributeConfig`
+ * should be used by subclasses to create base CustomAttributeConfig object inside the `bind()` method.
+ */
+abstract class BaseCustomAttribute implements ICustomAttribute {
```

Comment:
Same here. Let's drop 'Base'.

Unfortunately, if an exported class inherits this, this will also need to be exported.

```suggestion
abstract class CustomAttribute implements ICustomAttribute {
```

Reply:
Done ✅